timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/BatchingPaxosLatestSequenceCache.java

```diff
@@ -62,21 +58,35 @@
         try {
             return unsafeGetLatest(clients);
         } catch (RemoteException e) {
-            if (e.getError().errorName().equals(ERROR_NAME)) {
+            if (e.getError().errorName().equals(BatchPaxosAcceptor.CACHE_KEY_NOT_FOUND.name())) {
                 log.info("Cache key is invalid, invalidating cache");
-                cacheKey = null;
-                Set<Client> allClients = ImmutableSet.<Client>builder()
-                        .addAll(clients)
-                        .addAll(cachedEntries.keySet())
-                        .build();
-                cachedEntries.clear();
-                return unsafeGetLatest(allClients);
+                return handleCacheMiss(clients);
+            }
+
+            log.error("received remote exception that is not a cache key miss", e);
+            throw e;
+        } catch (RuntimeException e) {
+            // TODO(fdesouza): Remove this once we've moved to CJR properly and the above works.
+            if (e.getMessage().contains(BatchPaxosAcceptor.CACHE_KEY_NOT_FOUND.name())) {
+                log.info("Cache key is invalid, invalidating cache - using deprecated detection method");
+                return handleCacheMiss(clients);
             }
 
+            log.error("received unexpected runtime exception", e);
```

Comment:
should these be logged at `error`? Specifically I'm thinking about exceptions where one node is unreachable so the delegate could throw